WordPress/Classes/ViewRelated/Pages/PageListViewController.swift

```diff
@@ -318,115 +318,40 @@ class PageListViewController: AbstractPostListViewController, UIViewControllerRe
     // MARK: - Post Actions
 
     override func createPost() {
-        let navController: UINavigationController
+        let context = ContextManager.sharedInstance().mainContext
+        let postService = PostService(managedObjectContext: context)
+        let page = postService.createDraftPage(for: blog)
+        WPAppAnalytics.track(.editorCreatedPost, withProperties: ["tap_source": "posts_view"], with: blog)
+        showEditor(page: page)
+    }
 
+    fileprivate func editPage(_ apost: AbstractPost) {
+        WPAnalytics.track(.postListEditAction, withProperties: propertiesForAnalytics())
+        showEditor(page: apost)
+    }
+
+    fileprivate func showEditor(page: AbstractPost) {
         let filterIndex = filterSettings.currentFilterIndex()
         let editorSettings = EditorSettings()
-        if editorSettings.visualEditorEnabled {
-
-            if editorSettings.nativeEditorEnabled {
-                let context = ContextManager.sharedInstance().mainContext
-                let postService = PostService(managedObjectContext: context)
-                let page = postService.createDraftPage(for: blog)
-                let postViewController = AztecPostViewController(post: page)
-
-                navController = UINavigationController(rootViewController: postViewController)
-                navController.restorationIdentifier = AztecPostViewController.Restoration.navigationIdentifier
-                navController.restorationClass = AztecPostViewController.self
-                postViewController.onClose = { [weak self] changesSaved in
-                    if changesSaved {
-                        if let postStatus = postViewController.post.status {
-                            self?.updateFilterWithPostStatus(postStatus)
-                        }
-                    } else {
-                        self?.updateFilter(index: filterIndex)
-                    }
-                    navController.dismiss(animated: true, completion: nil)
-                }
-            } else {
-                let postViewController: EditPageViewController = EditPageViewController(draftFor: blog)
```

Comment:
the `draftFor:` method sets `ownsPost` within `EditPageViewController` to true, whereas `initWithPost:mode:` does not. The result is that cancelled new posts and pages now stick around as untitled drafts.